### PR TITLE
initial implementation

### PR DESCRIPTION
**Pull request by elbe0046**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (FreeMasen) [inline package.json:1]: I'd recomend starting with `npm init` which will fill this out more, including some nice default keys like `scripts` and whatnot. 

It also appears that you aren't encoding the actual dependencies here like `node-fetch` and `@aws-sdk/*`, using `npm install -s <module-name>` is a good way to have this automatically populated for you

File: src/index.js [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    let slackUserId = null;
+    for (const member of this.members) {
+      if (member.victorOpsUserId === victorOpsUserId) {
+        slackUserId = member.slackUserId;
+        break;
+      }
+    }
+
+    return slackUserId;
+  }
```
Comment (FreeMasen) [inline src/index.js]:
```suggestion
    let member = this.members.find(m => m.victorOpsUserId === victorOpsUserId);
    return member?.slackUserId
  }
```

File: src/index.js [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+class Record {
+  constructor(
+      victorOpsGroupId,
+      slackUserGroupId,
+      members,
+  ) {
+    this.victorOpsGroupId = victorOpsGroupId;
+    this.slackUserGroupId = slackUserGroupId;
+    this.members = members;
+  }
```
Comment (FreeMasen) [inline src/index.js:65]:
```suggestion
  /**
  * @param {string|number} victorOptsGroupId
  * @param {string|number} slackUserGroupId
  * @param {Member[]} members
  **/
  constructor(
      victorOpsGroupId,
      slackUserGroupId,
      members,
  ) {
    this.victorOpsGroupId = victorOpsGroupId;
    this.slackUserGroupId = slackUserGroupId;
    this.members = members;
  }
```

I'm only going to comment this function with type hints but by using [jsdoc](https://jsdoc.app) to provide type hints you can help anyone making future changes to get better auto completion.

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+  };
+  const resp = await fetch('https://slack.com/api/usergroups.users.update', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json; charset=utf8',
+      'Authorization': 'Bearer ' + botUserOAuthToken,
+    },
+    body: JSON.stringify(data),
+  });
+  return resp.json();
```
Comment (FreeMasen) [inline src/index.js]: You may want to make sure that `resp.ok == true`  here before trying to read the request body. Also `resp.json` returns a promise so you probably want `return await resp.json()`

File: src/index.js [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+  );
+  if (record == null) {
+    console.log('Rotation not found for victorOpsGroupId: ' + victorOpsGroupId);
+    return null;
+  }
+
+  const slackUser = record.getSlackUserId(victorOpsUserId);
+  const secret = await SecretsStore.getSecret(SLACK_TOKEN_SECRET_NAME, REGION);
+
+  if (slackUser == null) {
```
Comment (FreeMasen) [inline src/index.js:200]: Why would you get the secret before checking this?

File: src/index.js [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+  });
+  const docClient = DynamoDBDocument.from(dynamoClient, {
+    marshallOptions,
+  });
+  await docClient.put({
+    TableName: ROTATIONS_TABLE,
+    Item: record,
+  });
+
+  return {};
```
Comment (FreeMasen) [inline src/index.js]: Returning an empty table is odd?

File: src/index.js [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+      event.slackUserGroupId,
+      event.members.map((member) =>
+        new Member(member.victorOpsUserId, member.slackUserId)),
+  );
+
+  const marshallOptions = {
+    convertClassInstanceToMap: true,
+  };
+
+  const dynamoClient = new DynamoDBClient({
```
Comment (FreeMasen) [inline src/index.js]: I'm surprised this isn't designed to be a longer lived database connection. Is there a reason to create a new connection for each request as apposed to making requests while re-using the same client?

**Review by FreeMasen — APPROVED** [review]
Overall looks pretty good, just a few comments

File: package.json (hunk 1 above)
Comment (elbe0046) [reply package.json:1]: Very glad to learn about this, thanks!

File: src/index.js (hunk 2 above)
Comment (elbe0046) [reply src/index.js]: Much better

File: src/index.js (hunk 3 above)
Comment (elbe0046) [reply src/index.js:65]: This is great, I've updated to include this.  One piece I still need to figure out is how to properly link to imported pieces such as https://github.com/elbe0046/rotation-updater/commit/1846e57efc9fc73fe0eec3cea426cfb1dff794cf#diff-bfe9874d239014961b1ae4e89875a6155667db834a410aaaa2ebe3cf89820556R18.

File: src/index.js (hunk 5 above)
Comment (elbe0046) [reply src/index.js:200]: Great point, updated

File: src/index.js (hunk 7 above)
Comment (elbe0046) [reply src/index.js]: Ah nice, yeah I've updated to have these pieces as globals - is that what you had in mind, or is there a better way where they are still frozen and yet more narrowly scoped?

File: src/index.js (hunk 3 above)
Comment (FreeMasen) [reply src/index.js:65]: Those I believe should be resolving based on what is in your `node_modules` directory. You may need to configure your LSP server to know what types it should be aware of. In theory it should already be resolving [this d.ts file](https://github.com/aws/aws-sdk-js/blob/master/clients/dynamodb.d.ts) but you may need to update your configuration

File: src/index.js (hunk 7 above)
Comment (FreeMasen) [reply src/index.js]: No that is probably the best way to do it currently. The alternative would be to have some kind of database interface that could abstract these functions into methods that would own the database connection but I think that  would probably be overkill for what you are doing here.

**Review by FreeMasen — APPROVED** [review]

**Pull request merged**